src/System/Process/Pager.hs

```diff
@@ -64,7 +64,7 @@ pageWriter writer = do
  where
   cmdspecFromEnvVar = shell <$> MaybeT (lookupEnv "PAGER")
   cmdspecFromExeName =
-    fmap (\path -> proc path []) . MaybeT . findExecutable
+    fmap (`proc` []) . MaybeT . findExecutable
```

Comment:
I take your general point, although it might be said the original code of which this particular extract is a part is not easy to understand, full stop. I had to look up `System.Directory.findExecutable`, `Control.Monad.Trans.Maybe.MaybeT`, and the `Functor m => Functor (Maybe T m)` instance for `MaybeT` to understand it.

My general approach was (a) to remove each of the existing limitations in `.hlint.yaml` in turn, (b) to see what had been ignored in the existing Stack code base, and (c) if there were only a handful of HLint suggestions and all of them seemed to me to be improvements or, at least, not make things worse, to remove the existing limitation. At this site, I did not think it made things _worse_.

I think this is an application of HLint's "Avoid lambda using `infix`" rule. I'll reintroduce something to ignore that rule.